### PR TITLE
Rdiscrowd 3250 - Update project api to match UI

### PR DESCRIPTION
**Pull request by lsuttle**
*Issue number of the reported bug or feature request: Rdiscrowd 3250*

**Describe your changes**

1. Updated API to match project creation page requirements
   * Added ability to set password via api
   * Moved `description_from_long_description` function to utils and added it to project api pre-processing

2. Updated project validation to require the same fields as project web form

3. Updated tests to fit new requirements

**Testing performed**
Describe the testing you have performed to ensure that the bug has been addressed, or that the new feature works as planned.

**Additional context**
Add any other context about your contribution here.

### CODE REVIEW

File: pybossa/api/project.py [hunk 1]
```diff
@@ -60,7 +60,11 @@ def _preprocess_post_data(self, data):
 
 
     def _create_instance_from_request(self, data):
+        if 'password' not in data.keys():
+            raise BadRequest("password required")
```
Comment (kbecker42) [inline pybossa/api/project.py:64]: Do we have a test case to cover this exception?

Comment (lsuttle) [reply pybossa/api/project.py:64]: Not yet, only got the existing test cases to work, will be adding today.

Comment (lsuttle) [reply pybossa/api/project.py:64]: Now added.

**Comment by lsuttle** [comment]
@ywrsusan I think it does this, but let me know if this is missing something you expect. This is the json from a successful post:

```
{
  "info": {
    "data_classification": {
      "input_data": "L4 - public",
      "output_data": "L4 - public"
    },
    "data_access": [
      "L4"
    ],
    "passwd_hash": "pbkdf2:sha256:50000$lFTAk9vx$821100c557671dab572eddae8bdd392b9ae5922d674e6acd6e39fad013ca0827"
  },
  "updated": "2020-07-01T16:50:39.520602",
  "hidden": false,
  "description": "description",
  "short_name": "xxxx-project",
  "created": "2020-07-01T16:50:39.520575",
  "name": "XXXX Project",
  "webhook": null,
  "zip_download": true,
  "owner_id": 2,
  "id": 1,
  "featured": false,
  "email_notif": false,
  "allow_anonymous_contributors": true,
  "published": false,
  "secret_key": "2098d4eb-64b0-429e-8a64-6a1f768e23d1",
  "category_id": 1,
  "contacted": false,
  "long_description": "Long Description\\n================",
  "owners_ids": [
    2
  ]
}
```

**Comment by ywrsusan** [comment]
> @ywrsusan I think it does this, but let me know if this is missing something you expect. This is the json from a successful post from one of the tests:
> ...

That's very cool. Thanks for confirming.

File: pybossa/repositories/project_repository.py [hunk 2]
```diff
@@ -180,6 +180,22 @@ def _verify_data_classification(self, project):
...
+        if not project.short_name:
+            raise BadRequest("Short_name required")
+        if not project.description:
+            raise BadRequest("Description required")
+        kpi = project.info.get("kpi")
+        if kpi is None:
+            raise BadRequest("KPI required")
+        if not isinstance(kpi, (float, int)) or kpi > 120 or kpi < 0.1:
+            raise BadRequest("KPI must be value between 0.1 and 120")
+        if not project.info.get("product"):
```
Comment (kbecker42) [inline pybossa/repositories/project_repository.py:195]: Could we also check if the value for product and subproduct exists in the dictionary config value?

File: test/factories/project_factory.py [hunk 3]
```diff
@@ -31,6 +31,12 @@ def _create(cls, model_class, *args, **kwargs):
             project.set_password('hello')
         if 'data_classification' not in project.info:
             project.info['data_classification'] = dict(input_data="L4 - public", output_data="L4 - public")
+        if 'kpi' not in project.info:
+            project.info['kpi'] = 0.5
+        if 'product' not in project.info:
+            project.info['product'] = 'exists'
```
Comment (kbecker42) [inline test/factories/project_factory.py:37]: We may need to update this to a matching value from the config (product `abc`, and subproduct `def`).

**Pull request merged**